docs/tutorials/picorv32_ram.rst

```diff
@@ -0,0 +1,244 @@
+            mem_ready <= mem_valid;
+
+        // (Signals have the same name as the picorv32 module: use '.*' to autofill)
+        picorv32 rv32_soc (
+          .*
+        );
+
+        // SRAM with always-active chip select and write control bits.
+        sky130_sram_2kbyte_1rw1r_32x512_8 sram (
+            .clk0(clk),
+            .csb0('b0),
+            .web0(!(mem_wstrb != 0)),
+            .wmask0(mem_wstrb),
+            .addr0(mem_addr),
+            .din0(mem_wdata),
+            .dout0(mem_rdata),
+            .clk1(clk),
+            .csb1('b1),
+            .addr1('b0),
+            .dout1()
+        );
+    endmodule
+
+Finally, your core build script will need to be updated to import the new SRAM Library, and specify some extra parameters such as die size and macro placement:
+
+.. code-block:: python
+    :caption: <project_dir>/picorv32_top.py
+
+    import siliconcompiler
+
+    design = 'picorv32_top'
+    die_width = 1000
+    die_height = 1000
+
+    chip = siliconcompiler.Chip(design)
+    chip.load_target('skywater130_demo')
+
+    # Set input source files.
+    chip.input(f'{design}.v')
+    chip.input('picorv32.v')
+    chip.input('sram/sky130_sram_2k.bb.v')
+
+    # Set clock period, so that we won't need to provide an SDC constraints file.
+    chip.clock('clk', period=10)
+
+    # Set die outline and core area.
+    chip.set('constraint', 'outline', [(0,0), (die_width, die_height)])
+    chip.set('constraint', 'corearea', [(10,10), (die_width-10, die_height-10)])
+
+    # Setup SRAM macro library.
+    from sram import sky130_sram_2k
+    chip.use(sky130_sram_2k)
+    chip.add('asic', 'macrolib', 'sky130_sram_2k')
+
+    # SRAM pins are inside the macro boundary; no routing blockage padding is needed.
+    chip.set('tool', 'openroad', 'task', 'route', 'var', 'grt_macro_extension', '0')
+    # Disable CDL file generation until we can find a CDL file for the SRAM block.
+    chip.set('tool', 'openroad', 'task', 'export', 'var', 'write_cdl', 'false')
+
+    # Place macro instance.
+    chip.set('constraint', 'component', 'sram', 'placement', (500.0, 250.0, 0.0))
+    chip.set('constraint', 'component', 'sram', 'rotation', 270)
+
+    # Build on a remote server.
+    chip.set('option', 'remote', True)
+    chip.run()
+
+With all of that done, your project directory tree should look something like this::
+
+    <rundir>
+    ├── sram
+    │ ├── sky130_sram_2k.bb.v
+    │ ├── sky130_sram_2k.py
+    │ ├── sky130_sram_2kbyte_1rw1r_32x512_8.gds
+    │ └── sky130_sram_2kbyte_1rw1r_32x512_8.lef
+    ├── picorv32.py
+    ├── picorv32.v
+    ├── picorv32_top.py
+    └── picorv32_top.v
+
+Your ``picorv32_top.py`` build script should take about 20 minutes to run on the cloud servers if they are not too busy, with most of that time spent in the routing task. As with the previous designs, you should see periodic updates on its progress, and you should receive a screenshot and metrics summary once the job is complete:
```

Comment:
I would probably also replace "periodic" here with every 30 seconds to be explicit so as to set appropriate user expectations.